Add views

manototh opened the pull request:
Preview: https://axiom-mano-views.mintlify.app/query-data/views

Prerequisite: https://github.com/axiomhq/docs/pull/264

tothmano committed:
Add new Datasets tab details, create map fields

c-ehrlich committed:
Update apl/data-types/map-fields.mdx

Co-authored-by: Mano Toth <71388581+tothmano@users.noreply.github.com>

tothmano committed:
Update with latest UI

manototh committed:
Merge branch 'main' into mano/datasets-tab-update

tothmano committed:
Update with Dom's text

tothmano committed:
Update map-fields.mdx

tothmano committed:
Update map-fields.mdx

tothmano committed:
Add views

benenharrington commented on query-data/datasets.mdx:
```diff
@@ -55,6 +56,38 @@ For more information, see [Virtual fields](/query-data/virtual-fields).
 
+## Views
+
+Views allow you to grant other users scoped access to your datasets. This means that instead of sharing the whole dataset with other users, you have the option to share only a filtered and transformed representation of the data. You define a view using a query that applies filters and transformations to your dataset, and then you only share the results of this query with another user.
```
I think this describes one use case for Views, but [there are certainly others](https://www.notion.so/axiomhq/Views-06f0d9bfc0ab4f89bd3393b6db265088). The goal of the suggestion below is to describe the functionality at a high level, in a way that's relevant for all the use cases we have in mind. Very open to other ways of approaching this.

```suggestion
Views allow you to apply commonly-used filters and transformations to your dataset, and to treat the resulting output like you would any other dataset. The concept of a View in Axiom is similar to the concept of a virtual table in a database.
```

benenharrington commented on query-data/datasets.mdx:
```diff
@@ -55,6 +56,38 @@ For more information, see [Virtual fields](/query-data/virtual-fields).
+### Display view
+
+1. Select a dataset.
+1. In the **Views** section on the right, click the view you want to display.
```
I'm not sure this section is of much value

benenharrington commented on query-data/datasets.mdx:
```diff
@@ -55,6 +56,38 @@ For more information, see [Virtual fields](/query-data/virtual-fields).
+        It’s enough to grant access to the view. You don’t need to grant access to the underlying dataset.
+    </Note>
+1. Click **Save**.
+
```
Similar the comment above, I think this is focussing too much on a particular use case of views.

tothmano committed:
Address suggestions

manototh commented:
@benenharrington Many thanks for this. I see your point and I've addressed your suggestions: https://axiom-mano-views.mintlify.app/query-data/views

I think it's worth keeping the procedures, especially the one about sharing views, because it explains something that isn't intuitive in our UI at the moment.

benenharrington commented on query-data/views.mdx:
```diff
@@ -0,0 +1,45 @@
+
+### Grant access to view
+
+Grant other users access to a view in the same way you grant access to individual datasets:
```
```suggestion
All views are available to users with organization-level read access to views. You can allow users without these permissions to access an individual view in the same way you grant access to individual datasets:
```

manototh replied:
Thanks @benenharrington!
Please feel free to commit this if my rewrite works for you.

benenharrington commented:
Looks great Mano!

Added one small suggestion to make it explicit that certain users will have access to a view as soon as it's created (similar to datasets). Feel free to ignore if you don't think it's necessary

manototh commented on query-data/views.mdx:
```diff
@@ -0,0 +1,45 @@
+
+1. Click <img src="/doc-assets/icons/settings.svg" className="inline-icon" alt="Settings icon" /> **Settings > Roles**.
+1. Click the role with which you want to share the view or create a new role.
+1. In the **Organization** section, enable read access to views.
```
@benenharrington This should be removed, right?

benenharrington replied:
Woops, yep!

benenharrington committed:
Update query-data/views.mdx

tothmano committed:
Update views.mdx

tothmano committed:
Merge branch 'main' into mano/views

manototh committed:
Merge branch 'main' into mano/views

benenharrington committed:
Merge branch 'main' into mano/views
